third_party/move/move-core/types/src/vm_status.rs

```diff
@@ -729,9 +729,15 @@ pub enum StatusCode {
     // Failed to resolve type due to linking being broken after verification
     TYPE_RESOLUTION_FAILURE = 2021,
     DUPLICATE_NATIVE_FUNCTION = 2022,
+    // code invariant error while handling delayed fields, should never happen,
+    // always indicates a code bug.
+    DELAYED_FIELDS_CODE_INVARIANT_ERROR = 2023,
```

Comment:
I think the name is a bit confusing especially with the current usage - you also use thy on resource read in respawned session for example? Should we have just a single speculative error for example?

Reply:
Comment is a bit off, asking about both fields

Reply:
DELAYED_FIELDS_CODE_INVARIANT_ERROR is definitely correct.

for the second one - not sure what's best - 
DELAYED_FIELDS_SPECULATIVE_ABORT_ERROR, 
BLOCK_STM_SPECULATIVE_ERROR,
SPECULATIVE_EXECUTION_ERROR
?

Reply:
I think it may be useful beyond delayed fields (and no real need yet for block stm to necessarily distinguish those), so I do like SPECULATIVE_EXECUTION_ERROR, or SPECULATIVE_EXECUTION_INCONSISTENCY_ERROR